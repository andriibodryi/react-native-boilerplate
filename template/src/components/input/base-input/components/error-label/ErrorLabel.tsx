import React from 'react';
import { View } from 'react-native';

import { Text } from '@src/components/typography';
import { useStyleConfig } from '@src/styles/useStyleConfig';

import { styles } from './error-label.presets';
import { ErrorLabelTypes } from './error-label.types';

const ErrorLabel: React.FC<ErrorLabelTypes> = ({ message }) => {
  const { colors } = useStyleConfig();

  const isErrorLabelHidden = !message;
  if (isErrorLabelHidden) {
    return null;
  }

  return (
    <View style={styles.container}>
      <Text
        preset="textXsRegular"
        style={[
          styles.message,
          {
            color: colors.red_dark,
          },
        ]}
      >
        {message}
      </Text>
    </View>
  );
};

export default ErrorLabel;
