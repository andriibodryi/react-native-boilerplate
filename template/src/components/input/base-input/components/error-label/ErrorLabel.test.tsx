import React from 'react';
import { render } from '@testing-library/react-native';

import ErrorLabel from './ErrorLabel';

describe('ErrorLabel', () => {
  it('should render correctly', () => {
    const { toJSON } = render(<ErrorLabel message="Dummy Message" />);

    expect(toJSON()).toMatchSnapshot();
  });
});
