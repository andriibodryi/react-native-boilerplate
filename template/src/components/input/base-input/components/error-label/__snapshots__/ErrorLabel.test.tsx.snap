// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorLabel should render correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "marginBottom": 16,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "color": "#4D4D4D",
          "justifyContent": "center",
        },
        Object {
          "borderRadius": 8,
          "height": 16,
          "width": 16,
        },
        false,
        Array [
          Object {
            "borderWidth": 1,
          },
          Object {
            "borderColor": "#E32837",
          },
        ],
        Object {
          "backgroundColor": "#FFFFFF",
        },
      ]
    }
  >
    <View>
      <SvgMock
        fill="#E32837"
        height={12}
        width={12}
      />
    </View>
  </View>
  <Text
    accessibilityLabel="Dummy Message"
    accessibilityRole="alert"
    style={
      Array [
        Object {
          "color": "#4D4D4D",
          "fontFamily": "Aspira-Regular",
          "fontSize": 16,
          "fontWeight": "400",
          "lineHeight": 20,
        },
        Array [
          Object {
            "fontFamily": "Aspira-Regular",
            "fontSize": 12,
            "fontWeight": "400",
            "lineHeight": 16,
          },
          Object {},
          Object {
            "color": "#4D4D4D",
          },
          Array [
            Object {
              "flexBasis": "85%",
              "marginLeft": 6,
            },
            Object {
              "color": "#E32837",
            },
          ],
        ],
      ]
    }
  >
    Dummy Message
  </Text>
</View>
`;
